gobblin-config-management/gobblin-config-client/src/main/java/gobblin/config/client/ConfigClientUtils.java

```diff
@@ -0,0 +1,137 @@
+    else {
+      URI relative = cs.getStoreURI().relativize(configKeyURI);
+      return getConfigKeyPath(relative.getPath());
+    } 
+  }
+  
+  /**
+   * Require input URI's scheme/authority name match ConfigStore's scheme/authority
+   * if the input URI's authority is present, return the input ConfigStore's root URI
+   * else, return the URI which only contains the scheme name
+   * 
+   * @param configKeyURI
+   * @param cs
+   * @return
+   */
+  public static URI getDefaultRootURI(URI configKeyURI, ConfigStore cs){
+    preconditionCheck(configKeyURI, cs);
+    
+    if(configKeyURI.getAuthority()!=null){
+      return cs.getStoreURI();
+    }
+    
+    // configKeyURI is missing authority/configstore root "etl-hdfs:///datasets/a1/a2"
+    try {
+      return new URI(configKeyURI.getScheme(), null, "/", null, null);
+    } catch (URISyntaxException e) {
+      // should not come here
+      throw new RuntimeException("Can not build URI based on " + configKeyURI);
+    }
+  }
+  
+  /**
+   * Build the {@link  ConfigKeyPath} based on the absolute/relative path
+   * @param input - absolute/relative file path
+   * @return      - {@link  ConfigKeyPath} corresponding to the input
+   */
+  public static ConfigKeyPath getConfigKeyPath(String input){
+    ConfigKeyPath result = SingleLinkedListConfigKeyPath.ROOT;
+    String[] paths = input.split("/");
+    for(String p: paths){
+      // in case input start with "/", some elements could be "", which should be skip
+      if(p.equals("")){
+        continue;
+      }
+      result = result.createChild(p);
+    }
+    return result;
+  }
+  
+  /**
+   * 
+   * @param configKeyPath - {@link ConfigKeyPath} object which represent the config node in input {@link ConfigStore}
+   * @param cs            - the corresponding ConfigStore
+   * @return              - the absolute URI which combine the ConfigStore root URI and input ConfigKeyPath
+   */
+  public static URI getAbsoluteURI(ConfigKeyPath configKeyPath, ConfigStore cs){
+    URI storeRoot = cs.getStoreURI();
+    String path = storeRoot.getPath();
+    if(path.endsWith("/")){
+      path = path + configKeyPath.getAbsolutePathString().substring(1);
+    }
+    else{
+      path = path + configKeyPath.getAbsolutePathString();
+    }
+    
+    try {
+      return new URI(storeRoot.getScheme(), storeRoot.getAuthority(), path , null, null);
+    } catch (URISyntaxException e) {
+      // should not come here
+      throw new RuntimeException("Can not build URI based on " + configKeyPath);
+    }
+  }
+  
+  public static Collection<URI> getAbsoluteURI(Collection<ConfigKeyPath> configKeyPaths, ConfigStore cs){
+    Collection<URI> result = new ArrayList<>();
+    for(ConfigKeyPath p: configKeyPaths){
+      result.add(getAbsoluteURI(p, cs));
+    }
+    
+    return result;
+  }
+  
+  private static void preconditionCheck(URI configKeyURI, ConfigStore cs){
```

Comment:
Can you have a clearer name on what this is checking? Like `checkMatchingSchemeAndAuthority`?